gprofiler/metadata/system_metadata.py

```diff
@@ -194,15 +205,39 @@ class SystemInfo:
     spawn_uptime_ms: float
 
 
+def get_windows_network_details() -> Tuple[str, str]:
+    IPV4_PATTERN = r"^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$"
+    LOOPBACK = "127.0.0.1"
+    ADDR_KEY = "addr"
+    MAC_KEY = -1000
+    for iface in netifaces.interfaces():
+        mac_address = None
+        for key, addresses in netifaces.ifaddresses(iface).items():
+            for address in addresses:
+                if key == MAC_KEY:
+                    mac_address = address[ADDR_KEY]
+                    continue
+                if re.match(IPV4_PATTERN, address[ADDR_KEY]) is not None and address[ADDR_KEY] != LOOPBACK:
+                    return cast(Tuple[str, str], (mac_address, address[ADDR_KEY]))
```

Comment:
If you expect `mac_address` to be set by this point, it's nicer to add an explicit `assert mac_address is not None`. You can also assert that both are really `str`s. Generally I prefer `assert`s to `cast`s because the former prove their correctness in runtime. Use `cast` only when it's truly very obvious from the code.

Here you can use a utility `assert_cast`, so you'll have instead
```suggestion
                    return (assert_cast(str, mac_address), assert_cast(str, address[ADDR_KEY]))
```

Reply:
Done